pkg/controller/cluster/cluster_reconcile_test.go

```diff
@@ -0,0 +1,179 @@
+package cluster
+
+import (
+	"context"
+	"testing"
+
+	synv1alpha1 "github.com/projectsyn/lieutenant-operator/pkg/apis/syn/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// testSetupClient returns a client containing all objects in objs
+func testSetupClient(objs []runtime.Object) (client.Client, *runtime.Scheme) {
+	s := scheme.Scheme
+	s.AddKnownTypes(synv1alpha1.SchemeGroupVersion, objs...)
+	return fake.NewFakeClient(objs...), s
+}
+
+func TestGitRepoCreation(t *testing.T) {
+
+	name := "test-cluster"
+	namespace := "test"
+
+	cluster := &synv1alpha1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: synv1alpha1.ClusterSpec{
+			TenantRef: &synv1alpha1.TenantRef{
+				Name:      "test-tenant",
+				Namespace: namespace,
+			},
+			GitRepoTemplate: &synv1alpha1.GitRepoTemplate{
+				Spec: synv1alpha1.GitRepoSpec{
+					RepoName: "test",
+				},
+			},
+		},
+	}
+
+	objs := []runtime.Object{
+		&synv1alpha1.GitRepo{},
+		cluster,
+	}
+
+	cl, s := testSetupClient(objs)
+
+	c := &ReconcileCluster{client: cl, scheme: s}
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	_, err := c.Reconcile(req)
+	if err != nil {
+		t.Fatalf("reconcile: (%v)", err)
```

Comment:
As I understood it, you'd use `t.Errorf()` instead.
Also the https://github.com/stretchr/testify package could come in handy to use `assert.NoError(t, err)` for example.

Reply:
Will have a look at testify

Reply:
And also try to do the tests table driven.